LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/JsonHandlers/LnsStationConfiguration.cs

```diff
@@ -174,16 +174,17 @@ select JoinEui.Parse(eui))
                               JsonReader.Property("freq_range", from r in JsonReader.Array(JsonReader.UInt32())
                                                                 select (new Hertz(r[0]), new Hertz(r[1]))),
                               JsonReader.Property("DRs",
-                                  JsonReader.Array(from e in JsonReader.Tuple(JsonReader.Either(from n in JsonReader.UInt32().Validate(n => n == 0)
+                                  JsonReader.Array(from e in JsonReader.Tuple(JsonReader.Either(from n in JsonReader.Int32().Validate(n => n == 0)
```

Comment:
Just like there's `FskSpreadingFactor` defined as an exception:

https://github.com/Azure/iotedge-lorawan-starterkit/blob/00ba2d439a353477adb23dacca73cbc6db32f9c1/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/JsonHandlers/LnsStationConfiguration.cs#L161

You could introduce:

```c#
private const SpreadingFactor SkipSpreadingFactor = (SpreadingFactor)(-1);
```

Then the first query in the `JsonReader.Either` here could be changed to allow 0 or -1 for the spreading factor:

```c#
from n in JsonReader.Int32().Validate(n => n is 0 or -1)
select (SpreadingFactor)n
```

You can revert the following change and update the `select`.